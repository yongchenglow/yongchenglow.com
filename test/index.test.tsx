import renderer from 'react-test-renderer';
import Index from '@/src/pages';

describe('Index', () => {
  it('renders the html we want', async () => {
    const component = renderer.create(<Index />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
