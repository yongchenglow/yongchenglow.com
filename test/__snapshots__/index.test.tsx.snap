// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders the html we want 1`] = `
<div
  className="MuiBox-root css-bx7oc5"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular css-1cndkqf-MuiToolbar-root"
      >
        <div
          className="MuiBox-root css-178gejk"
        >
          <a
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            <button
              className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-4iq57c-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Home
            </button>
          </a>
          <a
            href="/about"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            <button
              className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-4iq57c-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              About
            </button>
          </a>
          <a
            href="/blog"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            <button
              className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-4iq57c-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Blog
            </button>
          </a>
        </div>
      </div>
    </div>
  </header>
  <div
    className="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
  >
    <div
      className="MuiBox-root css-16j30i3"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular css-15v5bzx-MuiAvatar-root"
      >
        <img
          alt="Low Yong Cheng"
          className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="/img/YongCheng.jpg"
        />
      </div>
      <div
        className="MuiBox-root css-164r41r"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
        >
          Hello and welcome to my webpage!
        </h4>
      </div>
    </div>
    <div
      className="MuiBox-root css-14fwub4"
    >
      <div
        className="MuiBox-root css-1qm1lh"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Check out my latest content
        </h5>
      </div>
      <div
        className="MuiBox-root css-1l4w6pd"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1o5cq17-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h6
              className="MuiTypography-root MuiTypography-h6 css-14y9ogl-MuiTypography-root"
            >
              Setting up your Project
            </h6>
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              What is the proper way to setup a group project? How do we make the best our of our IDE to increase code quality and productivity?
            </p>
          </div>
          <div
            className="MuiCardActions-root MuiCardActions-spacing css-u6qva2-MuiCardActions-root"
          >
            <a
              href="/blog/6"
              onClick={[Function]}
              onMouseEnter={[Function]}
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <button
                className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                disabled={false}
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                Read Now
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root css-14fwub4"
    >
      <div
        className="MuiBox-root css-19idom"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Who am I?
        </h5>
      </div>
      <div
        className="MuiBox-root css-1359uyy"
      >
        I am Yong Cheng, sometimes my friends call me YC for short. I am a Web Software Engineer currently working at Glints and I teach at Le Wagon Singapore part time. Check out some of my articles that I write as I try to share with others about my software engineering journey. You can find out more about me at my about page.
      </div>
    </div>
  </div>
  <footer
    style={
      Object {
        "marginTop": "auto",
      }
    }
  >
    <div
      className="MuiBox-root css-405ck5"
    >
      <div
        className="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container css-1xcs9jg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
          >
            <a
              className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-195ne0f-MuiTypography-root-MuiLink-root"
              href="https://www.linkedin.com/in/yong-cheng-low/"
              onBlur={[Function]}
              onFocus={[Function]}
              target="_blank"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LinkedInIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                />
              </svg>
            </a>
            <a
              className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-195ne0f-MuiTypography-root-MuiLink-root"
              href="https://github.com/yongchenglow"
              onBlur={[Function]}
              onFocus={[Function]}
              target="_blank"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="GitHubIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                />
              </svg>
            </a>
            <a
              className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-195ne0f-MuiTypography-root-MuiLink-root"
              href="https://www.instagram.com/yclow88/"
              onBlur={[Function]}
              onFocus={[Function]}
              target="_blank"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="InstagramIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
                />
              </svg>
            </a>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
          >
            © 2022 Copyright Yong Cheng Low
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 css-wg4xta-MuiGrid-root"
          >
            <a
              className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1wlbpa1-MuiTypography-root-MuiLink-root"
              href="mailto:lowyongcheng@hotmail.com"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-dzp38i-MuiSvgIcon-root"
                data-testid="EmailIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                />
              </svg>
               
              lowyongcheng@hotmail.com
            </a>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
